### ISSUE TITLE
FakeItEasy0003 false positive when returning argument constraint from helper method

### ISSUE THREAD

**Issue by thomaslevesque** [FakeItEasy/FakeItEasy#1076]
For instance this code doesn't compile because of FakeItEasy0003 error:

```
static int EvenNumber()
{
    return A<int>.That.Matches(i => i % 2 == 0);
}
```

Calling this method outside a call specification would throw, but calling it in a call specification is valid and works.

**Comment by thomaslevesque** [comment]
There's no way for the analyzer to detect this scenario; the best we can do is reduce the diagnostic severity to Warning.

**Comment by blairconrad** [comment]
Fixed by #1077, no matter what it thinks.

**Comment by thomaslevesque** [comment]
Fixed in release 3.3.1
